Update `gpg publish` to work with mirror arguments

douglasjacobsen opened the pull request:
This PR updates the `gpg publish` command to work with the mirror
arguments, when trying to push keys to a mirror.

douglasjacobsen committed:
Update `gpg publish` to work with mirror arguments

This commit updates the `gpg publish` command to work with the mirror
arguments, when trying to push keys to a mirror.

douglasjacobsen commented:
@opadron Can you review this for me? You had a comment in the code about doing this, so it'd be good to see if this is how you think it should be done.

tgamblin commented:
@opadron can you review this?

douglasjacobsen commented:
Gentle poke on this. Let me know if you have any comments so I can make required modifications.

opadron reviewed (dismissed):
This should be fine for now, but once #17547 lands (if I ever get around to finishing it!).  We'll need to update this handling code.

tgamblin commented:
talked to @douglasjacobsen -- he's going to add a test and then I think this is great!

douglasjacobsen committed:
Add tests for GPG publish

This commit adds test for publishing GPG keys and rebuilding
the key index within a mirror.

douglasjacobsen commented:
I added 3 tests (one for each mirror argument). So, I think this should be good to go, but let me know if you want any more tests to be added.

(This should also now kind of test `--rebuild-index` which wasn't covered before, but I'm not actually validating that the index is correct, just that it generates an index)
